src/frontend/js/hooks/useUnionResource/utils/syncIntegrityCount.ts

```diff
@@ -0,0 +1,122 @@
+import { MessageDescriptor } from 'react-intl';
+import { PaginatedResourceQuery } from 'types/Joanie';
+import { FetchEntityData, fetchEntities } from './fetchEntities';
+import { FetchEntityBaseArgs, QueryConfig } from './fetchEntity';
+import { hasIntegrity } from './hasIntegrity';
+import { log } from './log';
+
+interface SyncIntegrityCountArgs<DataA, DataB, FiltersA, FiltersB> extends FetchEntityBaseArgs {
+  stack: (DataA | DataB)[];
+  integrityCount: number;
+  newIntegrityCount: number;
+  queryAConfig: QueryConfig<DataA, FiltersA>;
+  queryBConfig: QueryConfig<DataB, FiltersB>;
+}
+interface SyncIntegrityCountSuccess<DataA, DataB> {
+  totalCount: number;
+  page: number;
+  stack: (DataA | DataB)[];
+  integrityCount: number;
+  errorMessage?: undefined;
+}
+interface SyncIntegrityCountError<DataA, DataB> {
+  errorMessage: MessageDescriptor;
+  totalCount: number;
+  page: number;
+  stack: (DataA | DataB)[];
+  integrityCount: number;
+}
+
+export const syncIntegrityCount = async <
+  DataA extends FetchEntityData,
+  DataB extends FetchEntityData,
+  FiltersA extends PaginatedResourceQuery = PaginatedResourceQuery,
+  FiltersB extends PaginatedResourceQuery = PaginatedResourceQuery,
+>({
+  stack,
+  newIntegrityCount,
+  queryAConfig,
+  queryBConfig,
+  page,
+  perPage,
+  eofRef,
+  queryClient,
+}: SyncIntegrityCountArgs<DataA, DataB, FiltersA, FiltersB>): Promise<
+  SyncIntegrityCountSuccess<DataA, DataB> | SyncIntegrityCountError<DataA, DataB>
+> => {
+  log('We want integrity', newIntegrityCount);
+
+  let pageToLoad = page;
+  let previousStack = [...stack];
+  let currentStack: (DataA | DataB)[];
+  let totalCount;
+  let integrityCount;
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    pageToLoad += 1;
+    currentStack = [...previousStack];
+    log('Fetching page ' + pageToLoad + ' ...');
+
+    try {
+      // eslint-disable-next-line no-await-in-loop
+      const res = await fetchEntities<DataA, DataB, FiltersA, FiltersB>({
+        stack: currentStack,
+        queryAConfig,
+        queryBConfig,
+        page: pageToLoad,
+        perPage,
+        eofRef,
+        queryClient,
+      });
+      totalCount = res.totalCount || totalCount;
+      currentStack = res.stack;
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
+      return {
+        errorMessage: {
+          id: 'hooks.useOrdersEnrollments.errorGet',
+          description:
+            'Error message shown to the user when orders or enrollments fetch request fails.',
+          defaultMessage:
+            'An error occurred while fetching orders and enrollments. Please retry later.',
+        },
```

Comment:
Something I missed previously I think, but this message should be declared at the module scope through `defineMessages` in order to be extracted by formatjs.